astropy/utils/tests/test_data.py

```diff
@@ -1131,20 +1146,13 @@ def osraiser(dirnm, linkto, pkgname=None):
         ),
     ],
 )
-def test_read_unicode(filename):
-    contents = get_pkg_data_contents(os.path.join("data", filename), encoding="utf-8")
-    assert isinstance(contents, str)
-    contents = contents.splitlines()[1]
-    assert contents == "האסטרונומי פייתון"
-
-    contents = get_pkg_data_contents(os.path.join("data", filename), encoding="binary")
-    assert isinstance(contents, bytes)
-    x = contents.splitlines()[1]
-    expected = (
-        b"\xff\xd7\x94\xd7\x90\xd7\xa1\xd7\x98\xd7\xa8\xd7\x95\xd7\xa0\xd7\x95"
-        b"\xd7\x9e\xd7\x99 \xd7\xa4\xd7\x99\xd7\x99\xd7\xaa\xd7\x95\xd7\x9f"[1:]
-    )
-    assert x == expected
+def test_read_unicode(filename, encoding, expected_type, expected_lines):
+    contents = get_pkg_data_contents(os.path.join("data", filename), encoding=encoding)
+    assert type(contents) is expected_type
+    # using str.splitlines here for portability:
+    # newlines are decoded as different character sequences
+    # on Unix ('\n') and Windows ('\r\n')
```

Comment:
```suggestion
```

I don't think it matters if it is `\n` or `\r\n` when it comes to necessity of splitlines? The old code already had splitlines anyway.

For the record, on Windows 11 with Git Bash, I get `\n` and not `\r\n`. If you want Powershell confirmation, we will have to summon Duy again because I don't want to mess up my env by also installing conda on Powershell.

You can see the endlines like this even without invoke package data. This is what the function is really doing. So you can download `unicode.txt` into a local disk and run the following from `main`:

```python
>>> filename = "unicode.txt"
>>> from astropy.utils.data import get_readable_fileobj
>>> with get_readable_fileobj(filename, encoding="utf-8") as fd:
...     contents = fd.read()
>>> contents  # Oh, it's Hebrew according to Google translate
'האסטרונומי פייתון\n'
```

Reply:
> The old code already had splitlines anyway.

Indeed. This comment is just a testimony of our struggling to remind ourselves why `str.splitlines` is more portable than `str.split('\n')`. I'm not too comfortable with removing it as it would feel like getting back to square one where I (or others) would be tempted to refactor it for whatever reason (I'm reasonably certain I will forget about this oddity again).

Reply:
OK then I propose slight verbiage change:

```
    # Using splitlines() instead of split("\n") here for portability as
    # newlines can be represented differently in different OSes.
```

This way, it is clearer why it is considered more portable (from what?) and we don't have to argue how exactly newlines are represented in NIX vs Windows.

Reply:
I agree, that's better ! Thank you !